plugins/paper/src/main/java/com/playmonumenta/plugins/depths/rooms/RoomRepository.java

```diff
@@ -87,6 +89,12 @@ public DepthsRoom summonRoom(Location spawnPoint, DepthsRoomType roomType, Depth
 		int floor = party.getFloor();
 		// Exception case- ascension 10+ and players must still remove abilities
 		if (party.getRoomNumber() % 10 == 9 && party.getAscension() >= DepthsEndlessDifficulty.ASCENSION_ABILITY_PURGE && !party.isAscensionPurgeMet()) {
+			if(this.mRoomCleanseSpawned) {
+				// TODO: Find a better way to terminate room spawn
+				throw new IllegalStateException("Each player must remove an ability before moving on!");
```

Comment:
No reason to throw an exception here - would be best to
- Send a message to players (either chat or actionbar)
- Return null
- Make this method @Nullable
- Add a null check in the one place it is used, `DepthsManager#playerSelectedRoom`